client/web/src/enterprise/batches/BatchSpecNode.tsx

```diff
@@ -105,19 +108,143 @@ export const BatchSpecNode: React.FunctionComponent<React.PropsWithChildren<Batc
             </div>
             {isExpanded && (
                 <div className={styles.nodeExpandedSection}>
-                    <H4>Input spec</H4>
+                    <BatchSpecInfo spec={node} isLightTheme={isLightTheme} />
+                </div>
+            )}
+        </>
+    )
+}
+
+interface BatchSpecInfoProps {
+    spec: Pick<BatchSpecListFields, 'originalInput' | 'id' | 'files' | 'description'>
+    isLightTheme: boolean
+}
+
+interface BatchSpecFile {
+    isBinary: boolean
+    content: string
+    isSpecFile: boolean
+    name: string
+    id: string
+    size: number
+}
+
+export const BatchSpecInfo: React.FunctionComponent<BatchSpecInfoProps> = ({ spec, isLightTheme }) => {
+    const specFile: BatchSpecFile = {
+        isBinary: false,
+        content: spec.originalInput,
+        isSpecFile: true,
+        name: 'spec_file.yaml',
+        id: spec.id,
+        size: 0,
```

Comment:
size: 0 seems like a lie 😛 
Maybe:
```suggestion
        size: spec.OriginalInput.length,
```
(although it's not always a 1-1 mapping.. sigh)

Reply:
Lol. Fair.